### PR TITLE
Using the BagIt to provide Bag validation within IngestArchivalMediaBagJob::ArchivalMediaBagParser

### LINKED ISSUE #1152: validate bag before ingesting in IngestArchivalMediaBagJob

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Resolves #1152

### CODE REVIEW

File: spec/change_set_persisters/plum_change_set_persister_spec.rb [hunk 1]
```diff
@@ -859,4 +859,32 @@
...
+  context 'when persisting a bag' do
+    let(:bag_path) { Rails.root.join('spec', 'fixtures', 'bags', 'valid_bag') }
+    let(:resource) { FactoryBot.build(:archival_media_collection) }
+    let(:change_set_class) { ArchivalMediaCollectionChangeSet }
+    let(:change_set) { change_set_class.new(resource, bag_path: bag_path) }
+
+    before do
+      stub_pulfa(pulfa_id: 'C0652')
+      change_set.prepopulate!
+      change_set.validate(source_metadata_identifier: 'C0652')
```
Comment (hackmastera) [inline spec/change_set_persisters/plum_change_set_persister_spec.rb]: it should create the collection if you didn't provide it; do you need this setup?

File: spec/fixtures/av/la_c0652_2017_05_bag/manifest-md5.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+ad34e40dc8137c01c7273ade9e38e5c6 data/32101047382401.xml
```
Comment (hackmastera) [inline spec/fixtures/av/la_c0652_2017_05_bag/manifest-md5.txt]: do these updates make the C0652 bag valid? that will be super helpful for using it as demo / dev data.

File: app/jobs/ingest_archival_media_bag_job.rb [hunk 3]
```diff
@@ -122,54 +137,80 @@ def barcode_with_part
...
           BARCODE_WITH_PART_REGEX.match(node.content)[1]
         end
     end
 
+    # Service Class for ingesting the bag as a procedure
     class Ingester
       attr_reader :collection, :bag, :user, :changeset_persister
+
+      # Constructor
+      # @param collection [Collection]
```
Comment (hackmastera) [inline app/jobs/ingest_archival_media_bag_job.rb]: [ArchivalMediaCollection]

File: spec/fixtures/av/la_c0652_2017_05_bag/manifest-md5.txt (hunk 2 above)
Comment (hackmastera) [reply spec/fixtures/av/la_c0652_2017_05_bag/manifest-md5.txt]: No need I believe you. thanks for fixing them.

File: spec/change_set_persisters/plum_change_set_persister_spec.rb (hunk 1 above)
Comment (jrgriffiniii) [reply spec/change_set_persisters/plum_change_set_persister_spec.rb]: For the `PlumChangeSetPersister`, I believe so, I've found the following when I attempt to pass an `ArchivalMediaCollectionChangeSet` without an instance of `ArchivalMediaCollection`:
```
Failure/Error: let(:change_set) { change_set_class.new(bag_path: bag_path) }
     
     NoMethodError:
       undefined method `state' for #<Hash:0x00007fc7c5e7b8e8>
```

Similarly, if I don't provide a `source_metadata_identifier` and do not stub a PULFA response:
```
Failure/Error: collection_component: post_save_resource.source_metadata_identifier.first,
     
     NoMethodError:
       undefined method `first' for nil:NilClass
     # ./app/change_set_persisters/plum_change_set_persister/ingest_bag.rb:18:in `run'
     # ./app/change_set_persisters/plum_change_set_persister.rb:171:in `block (2 levels) in after_save_commit'
```

But I have added an additional test for `IngestArchivalMediaBagJob` which shouldn't require that an instance of `ArchivalMediaCollection` have been created.

Comment (hackmastera) [reply spec/change_set_persisters/plum_change_set_persister_spec.rb]: I see; didn't realize we were in the persister spec.

**Comment by escowles** [comment]
Closing in favor of #1228

**Pull request closed without merging**